src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -672,6 +678,49 @@ def ok_to_stop(self,
             return HandleCommandResult(-errno.EBUSY, '', warn_message)
         return HandleCommandResult(0, warn_message, '')
 
+    def find_prometheus_credentials(self, multicluster_config: Dict[str, Any], url: str) -> Optional[Dict[str, Any]]:
+        for _, clusters in multicluster_config['config'].items():
+            for cluster in clusters:
+                prometheus_url = cluster['prometheus_url']
+                if prometheus_url:
+                    valid_url = prometheus_url.replace("https://", "").replace("http://", "")  # since target URLs are without scheme
+                    if valid_url == url:  # check if the target URL matches with the prometheus URL (without scheme) in the config
+                        return cluster['prometheus_access_info']
+        return None
+
+    def get_cluster_credentials(self, targets: List[str]) -> Dict[str, Any]:
+        clusters_credentials = {}
+        multi_cluster_config_str = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
+        try:
+            multi_cluster_config = json.loads(multi_cluster_config_str)
+            if multi_cluster_config:
+                for url in targets:
+                    credentials = self.find_prometheus_credentials(multi_cluster_config, url)
+                    if credentials:
```

Comment:
I think an else clause could be helpful here to debug issues. What happen if we cannot find the credentials? IMHO we should show some error

Reply:
Updated